### ISSUE TITLE
when you calculate loss, you may did not consider the padding part

### ISSUE THREAD

**Issue by Satan012** [Kyubyong/transformer#63]
the padding part is redundancy and should not be included in calculating the loss

**Comment by audier** [comment]
I am also confused，the code use: `tf.reduce_sum(keys, axis=-1)` to calculate the mask.
where keys =  Positional Encoding + Embedding layers.
for <pad>,
`
Positional Encoding  != 0
embedding = 0
keys != 0
`
so, how can it get the mask?

**Comment by gitfourteen** [comment]
@audier You may pass the non-position-encoded self.enc and self.dec as two more arguments via **multihead_attention** to _key_masks_ and _query_masks_, seperately.
